hw2/05-form.js

```diff
@@ -0,0 +1,59 @@
+function validateForm() {
+  const formatName = /[`!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?~0123456789]/;
+  const formatEmail = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
+  let fieldName = document.forms["form1"]["Username"].value;
+  let fieldEmail = document.forms["form1"]["email"].value;
+
+  let validBool = true;
+  if (fieldName.match(formatName)) {
+    window.alert("Please enter a valid Name");
+    validBool = false;
+  }
+  if (!fieldEmail.match(formatEmail)) {
+    window.alert("Please enter a valid Email");
+    validBool = false;
+  }
+  return validBool;
+}
+function resetForm() {
+  document.getElementById("form1").reset();
+}
+function submitForm() {
+  if (!validateForm()) {
+    return;
+  }
+  let name = document.getElementById("Username").value;
+  let email = document.getElementById("email").value;
```

Comment:
Instead of querying the DOM multiple times for the form inputs/values, please consider using form.elements to query the DOM for the form once [read about it here](https://eloquentjavascript.net/2nd_edition/18_forms.html)